legacy/builder/setup_build_properties.go

```diff
@@ -126,6 +127,17 @@ func (s *SetupBuildProperties) Run(ctx *types.Context) error {
 
 	buildProperties.Merge(ctx.PackageManager.CustomGlobalProperties)
 
+	// we check if the properties referring to secure boot have been set correctly.
+	if buildProperties.ContainsKey("build.keys.type") {
+		if buildProperties.Get("build.keys.type") == "public_keys" {
+			if !buildProperties.ContainsKey("build.keys.keychain") || !buildProperties.ContainsKey("build.keys.sign_key") || !buildProperties.ContainsKey("build.keys.encrypt_key") {
+				return errors.Errorf("%s core does not specify correctly default sign and encryption keys", ctx.BuildCore)
+			}
+		} else {
+			return errors.New("\"build.keys.type\" key only supports \"public_keys\" value for now")
+		}
+	}
+
```

Comment:
Yeah, this property is not used at the current time. But as I explained above, in the future we could add support for other approaches (like [a real keychain](https://github.com/arduino/arduino-cli/pull/1686/files#r826749584)). If we add something like this later, it could result in a breaking change. It was an idea by @cmaglie

Reply:
The default value allows it to be added later without breakage.

Reply:
https://github.com/arduino/arduino-cli/pull/1686/commits/80489c467a8ce7015cf2d8980d12ebd200859eff